### PR TITLE
Create StudentFilter Component

### LINKED ISSUE #55: Create 'StudentsFilter' component (child of 'StudentsFilters')
Create a filter component for the students filters bar

### PR DESCRIPTION
**Pull request by maxwolffe**
Fix #55 

Tests: Works locally, doesn't obviously break anything.

### CODE REVIEW

File: app/assets/javascripts/components/students_filter.es6.jsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  static get defaultProps() {
+    return {
+      isExpanded: false,
+      filterList: [],
+      selected: 'None',
+    };
+  }
+
+  handleExpand() {
+    this.setState({ isExpanded: !this.state.isExpanded })
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filter.es6.jsx]: Don't forget a semicolon after this `setState` call.

File: app/assets/javascripts/components/students_filter.es6.jsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  handleSelect(item) {
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filter.es6.jsx]: Change this name to `generateSelectHandler`.

File: app/assets/javascripts/components/students_filter.es6.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    function select() {
+      this.setState({ selected: item });
+    }
+    return select.bind(this)
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filter.es6.jsx]: You can just do
```
return function() {
  this.setState({ selected: true });
};
```

File: app/assets/javascripts/components/students_filter.es6.jsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  renderListItem(item) {
+    return (
+      <Clickable
+        func={this.handleSelect(item)}
+        styles={this.clickableStyles}>
+        <div>{item}</div>
+      </Clickable>
+      );
+    }
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filter.es6.jsx]: Indentation is off for lines 64 and 65.

File: app/assets/javascripts/components/students_filter.es6.jsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  renderListItems() {
+    return this.props.filterList.map(this.renderListItem.bind(this));
+  }
+
+  render() {
+    var style = Object.assign(
+      {},
+      this.styles.container,
+      this.state.isExpanded && this.styles.expanded
+    );
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filter.es6.jsx]: Nice use of `Object.assign`!

File: app/assets/javascripts/components/students_filter.es6.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    return (
+      <div style={style}>
+        <Clickable
+          func={this.handleExpand.bind(this)}
+          styles={this.clickableStyles}>
+          <span>{this.state.selected || this.props.selected}</span>
+          <i className="fa fa-angle-down"></i>
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filter.es6.jsx]: To follow our conventions, let's add curly braces around `"fa fa-angle-down"`.
So, just do `{'fa fa-angle-down'}` (also note we're using single quotes!).

File: app/assets/javascripts/components/students_filters.es6.jsx [hunk 7]
```diff
@@ -17,7 +18,14 @@ class StudentsFilters extends Component {
...
       <div style={this.styles.container}>
-        {'StudentsFilters'}
+        <StudentsFilter
+          filterList={['Warren', 'Anthony', 'Sonia']}
+          isExpanded={false}
+          selected={'Anthony'}/>
+        <StudentsFilter
+          filterList={['UnzUnz', 'Max', 'Cat']}
+          isExpanded={false}
+          selected={'Max'}/>
```
Comment (warrenshen) [inline app/assets/javascripts/components/students_filters.es6.jsx]: Don't forget the space before the `/>`.

**Comment by warrenshen** [comment]
@maxwolffe you have a couple more comments to outdate before we can get this pull in!

**Comment by maxwolffe** [comment]
Whoops. Thanks for the catch @warrenshen. I've made your changes.

**Pull request merged**